utils/blackbox.py

```diff
@@ -0,0 +1,363 @@
+from typing import Callable
+
+from algosdk.v2client import algod
+
+from graviton import blackbox
+
+from pyteal import (
+    Arg,
+    Btoi,
+    Bytes,
+    Expr,
+    Int,
+    Itob,
+    Len,
+    Log,
+    Mode,
+    ScratchVar,
+    Seq,
+    SubroutineFnWrapper,
+    TealType,
+    Txn,
+)
+
+# ---- Clients ---- #
+
+
+def algod_client(
+    algod_address="http://localhost:4001", algod_token="a" * 64
+) -> algod.AlgodClient:
+    """Instantiate and return Algod client object."""
+    return algod.AlgodClient(algod_token, algod_address)
+
+
+def algod_with_assertion():
+    algod = algod_client()
+    assert algod.status(), "algod.status() did not produce any results"
+    return algod
+
+
+# ---- API ---- #
+
+
+def mode_to_execution_mode(mode: Mode) -> blackbox.ExecutionMode:
+    if mode == Mode.Application:
+        return blackbox.ExecutionMode.Application
+    if mode == Mode.Signature:
+        return blackbox.ExecutionMode.Signature
+
+    raise Exception(f"Unknown mode {mode} of type {type(mode)}")
+
+
+def blackbox_pyteal(subr: SubroutineFnWrapper, mode: Mode) -> Callable[..., Expr]:
+    """Functor producing ready-to-compile PyTeal programs from annotated subroutines
+
+    Args:
+        subr: annotated subroutine to wrap inside program.
+            Note: the `input_types` parameters should be supplied to @Subroutine() annotation
+        mode: type of program to produce: logic sig (Mode.Signature) or app (Mode.Application)
+
+    Returns:
+        a function that called with no parameters -e.g. result()-
+        returns a PyTeal expression compiling to a ready-to-test TEAL program.
+
+    The return type is callable in order to adhere to the API of end-to-end unit tests.
```

Comment:
@tzaffi Optional - If possible, I prefer to avoid the _end-to-end_ terminology because we've otherwise adopted _blackbox_.  Feel welcomed to resolve if you prefer as is.

```suggestion
    The return type is callable in order to adhere to the API of blackbox tests.
```